views/pages/AdminPage.jsx

```diff
@@ -1,696 +1,201 @@
-import React, { Component } from 'react'
-import update from 'immutability-helper'
-import { Column, Table } from 'react-virtualized'
-import * as _ from 'lodash'
-import uuid from 'uuid'
-import moment from 'moment'
-
-import Settings from '@material-ui/icons/Settings'
-import Clear from '@material-ui/icons/Clear'
-
-import DialogActions from '@material-ui/core/DialogActions'
-import DialogTitle from '@material-ui/core/DialogTitle'
-import DialogContent from '@material-ui/core/DialogContent'
-import Dialog from '@material-ui/core/Dialog'
-import Button from '@material-ui/core/Button'
-import Subheader from '@material-ui/core/ListSubheader'
-import SelectField from '@material-ui/core/Select'
-import MenuItem from '@material-ui/core/MenuItem'
-import Checkbox from '@material-ui/core/Checkbox'
-import IconButton from '@material-ui/core/IconButton'
-import Typography from '@material-ui/core/Typography'
+import React, { useEffect, useState } from 'react'
+import { useForm, useFieldArray } from 'react-hook-form'
+import { useOutletContext } from 'react-router-dom'
 import NoSsr from '@material-ui/core/NoSsr'
 import Select from 'react-select'
-import TextField from '@material-ui/core/TextField'
-
+import { UsersModel, StudiesModel } from '../models'
 import { components } from '../forms/ControlledReactSelect/components'
-import getCounts from '../fe-utils/countUtil'
-import { fetchSubjects, fetchStudiesAdmin } from '../fe-utils/fetchUtil'
 import api from '../api'
-import basePathConfig from '../../server/configs/basePathConfig'
-const drawerWidth = 200
-const basePath = basePathConfig || ''
-const memberMenu = [
-  { value: 0, level: 'admin', text: 'System Admins' },
-  { value: 1, level: 'manager', text: 'Study Manager' },
-  { value: 2, level: 'member', text: 'Member' },
-]
+import AdminForm from '../forms/AdminForm'
+
+const AdminPageTwo = () => {
+  const usersKey = 'users'
```

Comment:
let's define this outside of the component so it doesn't get instantiated on every render.

Reply:
https://github.com/AMP-SCZ/dpdash/commit/1e4e37cd538b8fb695fa0cc8e622ee759d6c5515